File: TP1/File/File.c
#include <stdio.h>
#include <stdbool.h>

/**
 * Funcao principal
 * 
 * le um numero inteiro indicando quantos numeros reais serao
 * lidos, le essa quantidade de numeros reais e salva no arquivo
 * tmp.dat. Em seguida abre o arquivo tmp.dat e le os numeros de tras 
 * para frente printando na tela
 */
int main(){
    int n;
    double tmp;

    FILE* tmpFile = fopen("tmp.dat", "wb+");

    scanf("%d", &n);

    for(int i = 0; i < n; i++){
        scanf("%lf", &tmp);
        fwrite(&tmp, 8, 1, tmpFile);
    }

    for(int i = 0; i < n; i++){
        fseek(tmpFile, (n - i - 1) * 8, SEEK_SET);
        fread(&tmp, 8, 1, tmpFile);
        printf("%g\n", tmp);
    }

    fclose(tmpFile);
    return 0;
}

File: TP1/File/File.java
import java.io.IOException;
import java.io.RandomAccessFile;

public class File{
    /**
     * Metodo principal
     * 
     * le um numero inteiro indicando quantos numeros reais serao
     * lidos, le essa quantidade de numeros reais e salva no arquivo
     * tmp.dat. Em seguida abre o arquivo tmp.dat e le os numeros de tras 
     * para frente printando na tela
     */
    public static void main(String[] args) throws IOException{
        int n = MyIO.readInt();
        RandomAccessFile raf = new RandomAccessFile("tmp.dat", "rw");

        for(int i = 0; i < n; i++){
            double tmp = MyIO.readDouble();
            raf.writeDouble(tmp);
        }

        raf.close();

        raf = new RandomAccessFile("tmp.dat", "r");

        for(int i = 0; i < n; i++){
            raf.seek((n - i - 1) * 8);
            double tmp = raf.readDouble();
            int tmp2 = (int)tmp;

            if(tmp == tmp2)
                MyIO.println(tmp2);
            else
                MyIO.println(tmp);
        }

        raf.close();
    }
}
